_posts/2019/2019-12-16-how-to-enhance-an-application-using-firebase-auth-and-firebase-functions.md

```diff
@@ -0,0 +1,202 @@
+---
+layout: post
+title: How to enhance an application using Firebase Auth and Firebase Functions
+date: 2019-12-16
+header_image: public/firebase-post-2.jpg
+header_position: top
+header_overlay: true
+category: coding
+tags: ["firebase", "app", "cloud functions", "app development"]
+authors: ["Florian O."]
+about_authors: ["foellerich"]
+---
+
+In the [last part of this series](/blog/coding/how-to-bootstrap-an-application-with-authentication-using-firebase/){:target="_blank"}, we went through all the steps required to bootstrap an application with authentication.
+Our intermediate result was a login UI that can be used to distinguish users.
+Furthermore, we were able to give API access only to registered users.
+In this part, we'll take care of a database creation and security improvements.
+
+## Step 1 - Database creation
+
+Via the authentication service Firebase Auth, we receice tokens for different shops.
+Only the specific merchant of the shop should have access to this token.
+To do so, we first need to create a database in the Firebase console (Develop > Database).
+Then, we create a Cloud Firestore database.
+For fast results, we choose the test mode for now.
+We'll secure the data in a later step.
+Before we can start storing data, we need a collection for our users.
+That's why we create a collection called `users` in the Cloud Firestore UI.
+For the first document we need to provide, we use the document id `test`.
+The document does not need any properties yet.
+
+We'll now use the newly created collection `users` to save our tokens after the user logged in.
+First, we have to add the Firestore API to the `public/index.html`:
+
+```html
+  <script defer src="/__/firebase/7.2.0/firebase-firestore.js"></script>
+```
+
+Next, we can use it from our existing script after we `fetch` the access data via the API.
```

Comment:
I didn't really get what you mean with "it". Could you give me some insights Florian ?:)

Reply:
First, we have to add **the Firestore API** to the `public/index.html`:
Next, we can use **it** from our existing script after we `fetch` the access data via the API.